engine/unity5/Assets/Scripts/GUI/MainMenu.cs

```diff
@@ -66,25 +66,50 @@ public bool MyRemoteCertificateValidationCallback(System.Object sender, X509Cert
 
         private void Awake()
         {
-            WebClient client = new WebClient();
-            ServicePointManager.ServerCertificateValidationCallback = MyRemoteCertificateValidationCallback;
-            var json = new WebClient().DownloadString("https://raw.githubusercontent.com/Autodesk/synthesis/master/VersionManager.json");
-            VersionManager update = JsonConvert.DeserializeObject<VersionManager>(json);
-            updater = update.URL;
-
             string CurrentVersion = "4.2.0.1";
-            Auxiliary.FindObject(gameObject, "ReleaseNumber").GetComponent<Text>().text = "Version " + CurrentVersion;
 
-            var localVersion = new Version(CurrentVersion);
-            var globalVersion = new Version(update.Version);
+            if (CheckConnection())
+            {
+                WebClient client = new WebClient();
+                ServicePointManager.ServerCertificateValidationCallback = MyRemoteCertificateValidationCallback;
+                var json = new WebClient().DownloadString("https://raw.githubusercontent.com/Autodesk/synthesis/master/VersionManager.json");
+                VersionManager update = JsonConvert.DeserializeObject<VersionManager>(json);
+                updater = update.URL;
+
+               
+                Auxiliary.FindObject(gameObject, "ReleaseNumber").GetComponent<Text>().text = "Version " + CurrentVersion;
+
+                var localVersion = new Version(CurrentVersion);
+                var globalVersion = new Version(update.Version);
+
+                var check = localVersion.CompareTo(globalVersion);
 
-            var check = localVersion.CompareTo(globalVersion);
+                if (check < 0)
+                {
+                    Auxiliary.FindGameObject("UpdatePrompt").SetActive(true);
 
-            if (check < 0)
+                }
+            }
+            else
             {
-                Auxiliary.FindGameObject("UpdatePrompt").SetActive(true);
+                Auxiliary.FindObject(gameObject, "ReleaseNumber").GetComponent<Text>().text = "Version " + CurrentVersion;
+            }
+        }
 
-            }//client.DownloadFile(update.URL, @"C:\Users\t_moram\Downloads\Synthesis Installer.exe");
+        public static bool CheckConnection()
```

Comment:
This probably shouldn't be `static` as that'd make the method become defined at compile time, something which isn't necessary here.